tensorflow_model_optimization/python/core/sparsity_tf2/schedule_test.py

```diff
@@ -14,30 +14,361 @@
+
+    self._construct_schedule(schedule_type, 10, 1000, 1)
+    self._construct_schedule(schedule_type, 10, 1000, 10)
+
+  def _validate_drop_ratio(self, schedule_construct_fn):
+    # Should not be < 0.0
+    with self.assertRaises(ValueError):
+      schedule_construct_fn(-0.001)
+    with self.assertRaises(ValueError):
+      schedule_construct_fn(-1.0)
+    with self.assertRaises(ValueError):
+      schedule_construct_fn(-10.0)
+
+    # Should not be >= 1.0
+    with self.assertRaises(ValueError):
+      schedule_construct_fn(1.0)
+    with self.assertRaises(ValueError):
+      schedule_construct_fn(10.0)
+
+    schedule_construct_fn(0.0)
+    schedule_construct_fn(0.001)
+    schedule_construct_fn(0.5)
+    schedule_construct_fn(0.99)
+  
+
+  @parameterized.named_parameters(
+      {
+          'testcase_name': 'ConstantSparsity',
+          'schedule_type': 'constant_rate'
+      }, {
+          'testcase_name': 'CosineSchedule',
+          'schedule_type': 'cosine_decay'
+      }, {
+          'testcase_name': 'ExponentialSchedule',
+          'schedule_type': 'exponential_decay'          
+      }, {
+          'testcase_name': 'LRSchedule',
+          'schedule_type': 'lr_decay'
+      })
+  def testDropFractionValueIsValid(self, schedule_type):
+    begin_steps = [0, 1, 20]
+    end_steps = [25, -1]
+    for begin_step in begin_steps:
+      for end_step in end_steps:
+        if schedule_type == 'constant_rate':
+          # pylint: disable=unnecessary-lambda
+          self._validate_drop_ratio(lambda s: schedule.ConstantSchedule(s, begin_step, end_step))
+        elif schedule_type == 'cosine_decay':
+          # pylint: disable=unnecessary-lambda
+          self._validate_drop_ratio(lambda s: schedule.CosineSchedule(s, begin_step, end_step))
+        elif schedule_type == 'exponential_decay':
+          self._validate_drop_ratio(
+              lambda s: schedule.ExponentialSchedule(s, begin_step, end_step))
+        elif schedule_type == 'lr_decay':
+          self._validate_drop_ratio(
+              lambda s: schedule.LRSchedule(s, begin_step, end_step, 10, self.optimizer)) # TODO: None=optimizer arg to LR schedule test
+
+  # Tests to ensure begin_step, end_step, frequency are used correctly.
+
+  @keras_parameterized.run_all_keras_modes
+  @parameterized.named_parameters(
+      {
+          'testcase_name': 'ConstantSparsity',
+          'schedule_type': 'constant_rate'
+      }, {
+          'testcase_name': 'CosineSchedule',
+          'schedule_type': 'cosine_decay'
+      }, {
+          'testcase_name': 'ExponentialSchedule',
+          'schedule_type': 'exponential_decay'          
+      }, {
+          'testcase_name': 'LRSchedule',
+          'schedule_type': 'lr_decay'
+      })
+  def testPrunesOnlyInBeginEndStepRange(self, schedule_type):
+    decay = self._construct_schedule(schedule_type, 100, 200, 1)
+
+    # Before begin step
+    step_90 = tf.Variable(90)
```

Comment:
why are all these variables?

Reply:
We talked about this offline: the tests should avoid using `tf.variable`s except in the specific case of a test checking dynamicism when the drop fraction variable is changed